testcafeRunner.js

```diff
@@ -1,34 +1,50 @@
 import createTestcafe from 'testcafe';
 import { App } from './app';
 import routes from './app/routes';
+import { env } from './app/config';
 
 let testcafe;
-let server;
+const app = new App().createApp();
+app.use('/', routes);
+const server = app.listen('1234');
 
 const browserFromArgs = process.argv.slice(2, 3);
 const browserToRun = browserFromArgs.length > 0 ? browserFromArgs : 'chrome:headless';
 
 const testFromArgs = process.argv.slice(3, 4);
 const testsToRun = testFromArgs ? `**/*${testFromArgs}*/ui.test.js` : '**/*ui.test.js';
 
+let concurrency = 4;
+let stopOnFirstFail = true;
+let quarantineMode = true;
+if (env === 'pipeline' || browserFromArgs.length > 0) {
+  concurrency = 1;
+  stopOnFirstFail = false;
+  quarantineMode = false;
+}
+
+console.log(`Running tests in ${concurrency} threads\nstopOnFirstFail is ${stopOnFirstFail}\nquarantineMode is ${quarantineMode}`);
+
 createTestcafe('localhost')
   .then((tc) => {
     testcafe = tc;
 
-    const app = new App().createApp();
-    app.use('/', routes);
-
-    server = app.listen('1234');
-
-    return tc.createRunner()
-      .src([testsToRun])
+    return testcafe.createRunner()
+      .src(testsToRun)
       .browsers(browserToRun)
-      .concurrency(1)
+      .concurrency(concurrency)
       .reporter(['spec', {
         name: 'nunit',
         output: 'integration-test-report.xml',
       }])
-      .run();
+      .run({
+        selectorTimeout: 3000,
+        assertionTimeout: 1000,
+        pageLoadTimeout: 5000,
+        speed: 1,
+        quarantineMode,
```

Comment:
is it ok to just be putting `true`/`false` here rather than `{ key: value }`?

Reply:
Not sure I get what you mean

Reply:
never mind I am being stupid 🔨